### PR TITLE
An Alternative to "Use latest Azure Storage libraries" #2

### PR DESCRIPTION
**Pull request by martins-vds**
Copy of #36 

This PR is inspired by #33 and proposes a slightly different implementation.

### CODE REVIEW

**Comment by stebet** [comment]
@martins-vds This is looking good :) Can you fix the nullable warnings?

File: Functions/BlobStorageOptions.cs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+﻿namespace Functions
+{
+    public class BlobStorageOptions
+    {
+        public string BlobContainerName { get; set; }
+    }
+}
```
Comment (stebet) [inline Functions/BlobStorageOptions.cs:7]:
```suggestion
    public class BlobStorageOptions : IOptions<BlobStorageOptions>
    {
        public string BlobContainerName { get; set; }
        public BlobStorageOptions => this;
    }
}
```

**Review by stebet — CHANGES_REQUESTED** [review]

File: Functions.Tests/RangeTests.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Moq;
+using Xunit;
+
+namespace Functions.Tests
+{
+    public class RangeTests
```
Comment (stebet) [inline Functions.Tests/RangeTests.cs:14]: Can you add a test for `NotFound` as well?

File: Functions/BlobStorage.cs [hunk 3]
```diff
@@ -1,60 +1,67 @@
...
         /// Get a stream to the file using the hash prefix
         /// </summary>
         /// <param name="hashPrefix">The hash prefix to use to lookup the blob storage file</param>
         /// <returns>Returns a <see cref="BlobStorageEntry"/> with a stream to access the k-anonymity SHA-1 file</returns>
-        public async Task<BlobStorageEntry?> GetByHashesByPrefix(string hashPrefix)
+        public async Task<BlobStorageEntry?> GetHashesByPrefix(string hashPrefix, CancellationToken cancellationToken = default)
         {
             var fileName = $"{hashPrefix}.txt";
-            var blobClient = _blobContainerClient.GetBlobBaseClient(fileName);
+			var blobClient = _blobContainerClient.GetBlobClient(fileName);
```
Comment (stebet) [inline Functions/BlobStorage.cs:48]:
```suggestion
            var blobClient = _blobContainerClient.GetBlobClient(fileName);
```

**Review by stebet — CHANGES_REQUESTED** [review]

File: Functions/BlobStorageOptions.cs [hunk 4]
```diff
@@ -1,7 +1,8 @@
 ﻿namespace Functions
 {
-    public class BlobStorageOptions
+    public class BlobStorageOptions : IOptions<BlobStorageOptions>
     {
         public string BlobContainerName { get; set; }
+        public BlobStorageOptions => this;
```
Comment (stebet) [inline Functions/BlobStorageOptions.cs:6]: Whoops! Was supposed to be:
```suggestion
        public BlobStorageOptions Value => this;
```

**Review by stebet — CHANGES_REQUESTED** [review]

File: Functions/BlobStorageOptions.cs [hunk 5]
```diff
@@ -0,0 +1,8 @@
+﻿namespace Functions
```
Comment (stebet) [inline Functions/BlobStorageOptions.cs:1]:
```suggestion
using Microsoft.Extensions.Options;

﻿namespace Functions
```

**Review by stebet — CHANGES_REQUESTED** [review]

**Comment by stebet** [comment]
@martins-vds Can you rebase this PR on `main`after the latest merges so we can accept it :)

**Comment by martins-vds** [comment]
> @martins-vds Can you rebase this PR on the latest `main` so we can accept it :)

Done!

**Pull request merged**